lib/runtime-core/src/state.rs

```diff
@@ -86,8 +109,85 @@ pub enum MachineValue {
     TwoHalves(Box<(MachineValue, MachineValue)>), // 32-bit values. TODO: optimize: add another type for inner "half" value to avoid boxing?
 }
 
+impl BorshSerialize for MachineValue {
+    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        match self {
+            MachineValue::Undefined => writer.write_all(&(0 as u8).to_le_bytes())?,
+            MachineValue::Vmctx => writer.write_all(&(1 as u8).to_le_bytes())?,
+            MachineValue::VmctxDeref(v) => {
+                writer.write_all(&(2 as u8).to_le_bytes())?;
```

Comment:
For u8 values endianness doesn't matter.

Reply:
Good point

Reply:
replaced with `write(&[2u8])?` etc.